Thecarisma-syntax.md

```diff
@@ -0,0 +1,620 @@
+- Types can be automatically inferred in Ratio.
+    ```js
+    var todo = "Create a new programming language :)" // `todo` is inferred as `String`
+    ```
+
+    ```cpp
+    func add(a, b) {
+        return a + b
+    }
+
+    const sum = add(1, 1) // Usage of `add` inferred as `(Int, Int) -> Int` here.
+    ```
+
+- Subjects can be explicitly type-annotated.
+
+    ```js
+    var identification: Int = 60
+    ```
+
+- Functions can also be explicitly type-annotated.
+
+    ```swift
+    func add(a: Int, b: Int) -> Int {
+        return a + b
+    }
+    ```
+
+
+## DESTRUCTURING
+- Destructuring a tuple.
+
+    ```js
+    const (name, age) = ("John", 50)
+    ```
+
+- Destructuring a list.
+
+    ```js
+    const [..., last] = [1, 2, 3, 4, 5]
+    ```
+
+- Destructuring an object.
+
+    ```js
+    const {name, age} = john
+    ```
+
+## IF EXPRESSION
+- Conditional branches using `if`, `else if` and `else`.
+
+    ```swift
+    if cond1 {
+        print("if")
+    } else if cond2 {
+        print("else if")
+    } else {
+        print("else ")
+    }
+    ```
+
+- Checking for falsy values.
+
+    ```swift
+    if !complete { 
+        print("incomplete")
+    }
+    ```
+
+- Checking for Noneness and binding value to a name.
+
+    ```swift
+    if const stock_code = get_stock_code("") {
+        print(stock_code)
+    }
+    ```
+
+## FOR LOOP
+- A `for` loop iterates through an iterable and binds the value to a subject.
+
+    ```py
+    for i in 1:11 {
+        print(i)
+    }
+    ```
+
+- Destructuring in loop.
+
+    ```py
+    for (kind, number) in interesting_numbers {
+        print(kind, number)
+    }
+    ```
+
+- A `for` loop can be extended with a `where` condiition.
+    ```swift
+    for i in array where i%2 == 0 {
+        print(i)
+    }
+    ```
+
+- for loop must return a value if it an expression for subject assignment is nothing is returned the last value is returned.
+
+    ```js
+        const num = for i in 1:100 {
+            if i == 59 {
+                return i
+            }
+        }
+        print(num) //59
+    ```
+
+    ```js
+        const num = for i in array {
+            
+        }
+        print(num) //array[array.length-1]
+    ```
+
+- break cannot be used in a for loop that is expected to return a value but can be used to break out of the iteration if not return value is expected.
+
+    ```js
+        const num = for i in 1:100 {
+            if i == 59 {
+                break //Invalid. Error
+            }
+        }
+    ```
+
+    ```js
+        var result = 0
+        for i in 1:100 {
+            if i == 59 {
+                break //valid
+            } 
+            var += i
+        }
+    ```
+
+## EXPRESSION ORIENTED
+- Almost everything in Ratio (save a few constructs like impl, trait, type and import) is an expression!
+
+    ```js
+    var is_appcypher_crazy = (fave_hobby == "CountingBirds")
+    ```
+
+    ```rust
+    const name = (condition ? "John" -> "Peter" )
```

Comment:
For me, the `->` operator looks a bit confusing considering the fact it uses the same operator as the return statement.

At first glance, one would expect that it follows the same style of after the `->` operator, a return type follows immediately.

Reply:
I agree with @TNkemdilim on this. We should disambiguate symbols and the tokens. If it already means something before. It shouldn't mean something else.

Maka Safety....